### PR TITLE
update to es6 promises and remove when dependency

### PR DESCRIPTION
**Pull request by ccarruitero**

### CODE REVIEW

**Comment by willdurand** [comment]
That looks like something we want. What do you think @kumar303?

File: tests/test.amo-client.js [hunk 1]
```diff
@@ -1094,27 +1093,31 @@ class MockRequest {
       response = {};
       response.responseError = new Error("Response queue is empty");
     }
-    callback(response.responseError, response.httpResponse,
-             response.responseBody);
+
+    if (response.responseError instanceof Error) {
+      throw response.responseError;
```
Comment (kumar303) [inline tests/test.amo-client.js]: Does this still work the same as before? I'm not sure what was going on with the callbacks.

Comment (kumar303) [reply tests/test.amo-client.js]: In other words, do tests that rely on error responses still get the same behavior? To test it, you could try and make one of the tests fail that was relying on this mock.

**Comment by kumar303** [comment]
@ccarruitero this would need to be rebased on master to pick up the new `package.json`

File: tests/test.amo-client.js (hunk 1 above)
Comment (ccarruitero) [reply tests/test.amo-client.js]: @kumar303 sorry for delay.

There are some test that relay on the mock error response ([1](https://github.com/mozilla/sign-addon/blob/master/tests/test.amo-client.js#L359), [2](https://github.com/mozilla/sign-addon/blob/master/tests/test.amo-client.js#L876)), but let me know if you consider to add any other test.

**Pull request merged**